### ISSUE TITLE
TypeError: Cannot read property 'jsonTransportMode' of null

### ISSUE THREAD

**Issue by timarandras** [deepstreamIO/deepstream.io-client-js#501]
A simple client connection causes a TypeError:
```javascript
const { Deepstream } = require('@deepstream/server');
const DSClient = require('@deepstream/client');

const server = new Deepstream({
  connectionEndpoints: [
    {
      type: 'uws-websocket',
      options: {
        headers: ['cookie', 'user-agent', 'host', 'origin'],
      },
    },
  ],
});

server.start();

const client = DSClient('localhost:6020');
client.login();

client.rpc.provide('echo', (data, response) => {
  response.send(data);
});
```

```
/sandbox/node_modules/@deepstream/client/dist/connection/socket-factory.js:25
    var buildMessage = options.jsonTransportMode !== true ? message_builder_1.getMessage : function (message, isAck) { return JSON.stringify(__assign({}, message, { isAck: isAck })); };
                               ^
TypeError: Cannot read property 'jsonTransportMode' of null
    at Object.exports.socketFactory (/sandbox/node_modules/@deepstream/client/dist/connection/socket-factory.js:25:32)
    at Connection.createEndpoint (/sandbox/node_modules/@deepstream/client/dist/connection/connection.js:205:39)
    at new Connection (/sandbox/node_modules/@deepstream/client/dist/connection/connection.js:88:18)
    at new Client (/sandbox/node_modules/@deepstream/client/dist/client.js:57:31)
    at client (/sandbox/node_modules/@deepstream/client/dist/deepstream.js:5:12)
    at Object.<anonymous> (/sandbox/src/index.js:17:16)
    at Module._compile (internal/modules/cjs/loader.js:776:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:787:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
```
node -v = v12.9.0
@deepstream/server = 4.2.0
@deepstream/client = 4.1.1

Demo:
https://codesandbox.io/embed/jsontransportmode-error-w4s3l

**Comment by timarandras** [comment]
Seems to work with @deepstream/client = 4.1.0.

**Comment by xtreamtrader** [comment]
I am having the same issue using https://cdn.deepstream.io/js/client/latest/ds.min.js in browser

ds.min.js:1 Uncaught TypeError: Cannot read property 'jsonTransportMode' of null
    at Object.e.socketFactory (ds.min.js:1)
    at t.createEndpoint (ds.min.js:1)
    at new t (ds.min.js:1)
    at new e (ds.min.js:1)
    at o (ds.min.js:1)
    at store-6f6193df.js:720

@deepstream/client = 4.1.0. used to work.

**Comment by timarandras** [comment]
Worked around like this
```javascript
const { Deepstream } = require('@deepstream/server');
const DSClient = require('@deepstream/client');

const server = new Deepstream({
  connectionEndpoints: [
    {
      type: 'uws-websocket',
      options: {
        headers: ['cookie', 'user-agent', 'host', 'origin'],
      },
    },
  ],
});

server.start();

const client = DSClient('localhost:6020', {socketOptions: {
  jsonTransportMode: false
}});

async function start() {
  try {
    await client.login();    
    client.rpc.provide('echo', (data, response) => {
      response.send(data);
    });
  } catch (err) {
    console.error(err);   
  }
}

start();
```

Also forgot to await for login().

**Comment by yasserf** [comment]
Thank you for raising! I’ll be putting a default in and releasing ASAP

**Comment by yasserf** [comment]
Fixed in 4.1.2, sorry about that 🙈
